tests/lib/cli-engine.js

```diff
@@ -2877,4 +2889,48 @@ describe("CLIEngine", () => {
         });
     });
 
+    describe("mutability", () => {
+        describe("plugins", () => {
+            it("Loading plugin in one instance doesnt mutate to another instance", () => {
+                const filePath = getFixturePath("single-quoted.js");
+                const engine1 = cliEngineWithPlugins({
+                    cwd: path.join(fixtureDir, ".."),
+                    useEslintrc: false,
+                    plugins: ["example"],
+                    rules: { "example/example-rule": 1 }
+                });
+                const engine2 = new CLIEngine({
+                    cwd: path.join(fixtureDir, ".."),
+                    useEslintrc: false
+                });
+                const fileConfig1 = engine1.getConfigForFile(filePath);
+                const fileConfig2 = engine2.getConfigForFile(filePath);
+
+                // plugin
+                assert.strictEqual(fileConfig1.plugins[0], "example", "Plugin is present for engine 1");
+                assert.isUndefined(fileConfig2.plugins, "Plugin is not present for engine 2");
+            });
+        });
+
+        describe("rules", () => {
+            it("Loading rules in one instance doesnt mutate to another instance", () => {
+                const filePath = getFixturePath("single-quoted.js");
+                const engine1 = new CLIEngine({
+                    cwd: path.join(fixtureDir, ".."),
+                    useEslintrc: false,
+                    rules: { "example/example-rule": 1 }
+                });
+                const engine2 = new CLIEngine({
+                    cwd: path.join(fixtureDir, ".."),
+                    useEslintrc: false
+                });
+                const fileConfig1 = engine1.getConfigForFile(filePath);
+                const fileConfig2 = engine2.getConfigForFile(filePath);
+
+                // plugin
+                assert.strictEqual(fileConfig1.rules["example/example-rule"], 1, "example is present for engine 1");
+                assert.isUndefined(fileConfig2.rules["example/example-rule"], "example is not present for engine 2");
+            });
```

Comment:
In this test we are not loading a plugin, we are just testing in reference to rules being added. I just added the rules section above but i wanted to keep the test different and focused on one aspect.